src/cache.c

```diff
@@ -9,12 +9,11 @@
  */
 struct cache_entry *alloc_entry(char *path, char *content_type, void *content, int content_length)
 {
-    struct cache_entry *entry = malloc(sizeof(cache_entry));
-    // entry->content = malloc(sizeof(char) * strlen(content));
-    // entry->content = content;
-    entry->path = malloc(sizeof(char) * strlen(path));
+    printf("here1\n");
+    struct cache_entry *entry = malloc(sizeof(struct cache_entry));
```

Comment:
you're making a pointer, so you only need pointer size
`struct cache_entry *entry = malloc(sizeof(*entry));`